docs/source/en/agents.md

```diff
@@ -0,0 +1,570 @@
+
+```python
+from transformers import Tool
+
+class HFModelDownloadsTool(Tool):
+    pass
+```
+
+This class has a few needs:
+- An attribute `name`, which corresponds to the name of the tool itself. To be in tune with other tools which have a
+  performative name, we'll name it `model_download_counter`.
+- An attribute `description`, which will be used to populate the prompt of the agent.
+- `inputs` attribute. This is a dictionnary with keys `"type"`and `"description"`containing information to help the python interpreter make educated choices about the input,
+  and will allow for a gradio-demo to be spawned when we push our tool to the Hub.
+- An attribute `output_type` specifying the type of the ouput. 
+- A `__call__` method which contains the inference code. This is the code we've played with above!
+
+Here's what our class looks like now:
+
+```python
+from transformers import Tool
+from huggingface_hub import list_models
+
+class HFModelDownloadsTool(Tool):
+    name = "model_download_counter"
+    description = (
+        "This is a tool that returns the most downloaded model of a given task on the Hugging Face Hub. "
+        "It returns the name of the checkpoint."
+    )
+
+    inputs = {
+        "task": {
+            "type": str,
+            "description": "the task category (such as text-classification, depth-estimation, etc)",
+        }
+    }
+    output_type = str
+
+    def __call__(self, task: str):
+        model = next(iter(list_models(filter=task, sort="downloads", direction=-1)))
+        return model.id
+```
+
+We now have our tool handy. Save it in a file and import it from your main script. Let's name this file
+`model_downloads.py`, so the resulting import code looks like this:
+
+```python
+from model_downloads import HFModelDownloadsTool
+
+tool = HFModelDownloadsTool()
+```
+
+In order to let others benefit from it and for simpler initialization, we recommend pushing it to the Hub under your 
+namespace. To do so, just call `push_to_hub` on the `tool` variable (be sure to have created the space beforehand, and to be authenticated with a token that has read access!)
+
+```python
+tool.push_to_hub("{your_username}/hf-model-downloads")
+```
+
+You now have your code on the Hub! Let's take a look at the final step, which is to have the agent use it.
+
+### Initializing an agent that uses the tool
+
+We now have our tool that lives on the Hub which can be instantiated as such (change the user name for your tool):
+
+```python
+from transformers import load_tool
+
+model_download_tool = load_tool("m-ric/hf-model-downloads")
+```
+
+In order to use it in the agent, simply pass it to the agent initialization method.
+
+```python
+from transformers import CodeAgent
+
+agent = CodeAgent(llm_engine, tools=[model_download_tool])
+
+agent.run(
+    "Can you give me the name of the model that has the most downloads in the 'text-to-video' task on the Hugging Face Hub?"
+)
+```
```

Comment:
```suggestion
```python
from transformers import load_tool, CodeAgent

model_download_tool = load_tool("m-ric/hf-model-downloads")
agent = CodeAgent(llm_engine, tools=[model_download_tool])
agent.run(
    "Can you give me the name of the model that has the most downloads in the 'text-to-video' task on the Hugging Face Hub?"
)
```
```